gpjax/parameters.py

```diff
@@ -1,27 +1,51 @@
 import typing as tp
 import warnings
+from collections import namedtuple
 from copy import deepcopy
 
 import distrax as dx
 import jax
 import jax.numpy as jnp
+from chex import dataclass
 from jaxtyping import f64
 from tensorflow_probability.substrates.jax import distributions as tfd
 
 from .config import get_defaults
+from .utils import merge_dictionaries
 
 Identity = dx.Lambda(lambda x: x)
 
 
 ################################
 # Base operations
 ################################
-def initialise(obj) -> tp.Tuple[tp.Dict, tp.Dict, tp.Dict]:
+@dataclass
+class ParameterState:
+    """The state of the model. This includes the parameter set and the functions that allow parameters to be constrained and unconstrained."""
+
+    params: tp.Dict
+    trainables: tp.Dict
+    constrainers: tp.Dict
+    unconstrainers: tp.Dict
+
+    def unpack(self):
+        return self.params, self.trainables, self.constrainers, self.unconstrainers
+
+
+def initialise(model, key: jnp.DeviceArray, **kwargs) -> ParameterState:
     """Initialise the stateful parameters of any GPJax object. This function also returns the trainability status of each parameter and set of bijectors that allow parameters to be constrained and unconstrained."""
-    params = obj.params
+    params = model._initialise_params(key)
+    # for k, v in kwargs.items():
+    #     params[k] = merge_dictionaries(params[k], v)
     constrainers, unconstrainers = build_transforms(params)
     trainables = build_trainables(params)
-    return params, trainables, constrainers, unconstrainers
+    state = ParameterState(
+        params=params,
+        trainables=trainables,
+        constrainers=constrainers,
+        unconstrainers=unconstrainers,
+    )
+    return state
```

Comment:
I like this. Passing a key makes good sense.

Do we have an example of overwriting default parameter values, or initialising them according them to a key? i.e., unless I'm missing something, the method `._initialise_params(key)` seems to default to fixed values, independent of the key for the objects I've looked at thus far.

Reply:
Yes - in latest commit.